app/scripts/modules/core/src/pipeline/filter/executionFilter.service.ts

```diff
@@ -368,4 +371,49 @@ export class ExecutionFilterService {
     ExecutionState.filterModel.asFilterModel.clearFilters();
     ExecutionState.filterModel.asFilterModel.applyParamsToUrl();
   }
+
+  public static getRelation() {
+    return flatten(this.getRelations());
+  }
+
+  public static getRelations(): any {
```

Comment:
It looks like you are trying to determine all the `keys` that have each `relationVal`. One thing you can do is this:
```js
import { forIn } from 'lodash'; // at top of file

const relationValuesToKeys = {};
forIn(relation, (value, key) => {
    const relationVal = relationValuesToKeys[value];
    if (relationVal) {
       relationValuesToKeys[value] = relationVal.push(key)
    } else {
      relationValuesToKeys[value] = [key];
    }
 }
)

return Object.values(relationValuesToKeys);
```